storage/store.go

```diff
@@ -1587,20 +1602,21 @@ func (s *Store) processRaft() {
 					panic(err) // TODO(bdarnell)
 				}
 			}
+			s.processRaftMu.Unlock()
```

Comment:
You are introducing a dependency here between the locks processRaftMu -> mu. How are we confident this doesn't cause a deadlock?

Separately, is there value to abstracting this into a separate function that returns replicas?